### PR TITLE
feat(router): add poll ability in external 3ds authorization flow

### PR DESCRIPTION
**Pull request by sai-harsha-vardhan**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
add poll ability in external 3ds authorization flow

1. restrict authorization (post-auth) only when pull mechanism for 3ds is enabled for a merchant (this config is fetched from mca)
2. when status is in requires_customer_action, in generate_response create a poll_id in redis as pending, send the poll_status post message to client for polling along with poll configs
3. In webhooks flow, once the payment is authorized set the poll_id to completed so that client can end the process and return the result

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## How did you test it?
Tested Manually
1. Santiy of 3DS payments with redirection flow
2. Sanity of 3DS payments with complete authorize flow
3. Sanity of External 3ds authentication flow, both friction and frictionless through threedsecureio <> checkout connectors
Webhooks and Polling changes can be verified in SDK, after netcetera integration

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/utils.rs [hunk 1]
```diff
@@ -1090,6 +1094,138 @@ pub async fn get_profile_id_from_business_details(
...
+                    .get_redis_conn()
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("Failed to get redis connection")?;
+                // Set poll_id in redis to allow the fetch status of poll through retrieve_poll_status api from client
+                redis_conn
+                    .set_key_with_expiry(
+                        &poll_id,
+                        PollStatus::Pending.to_string(),
+                        // 15 minutes = 900 seconds
+                        900,
```
Comment (Narayanbhat166) [inline crates/router/src/core/utils.rs:1132]: move this to consts

File: crates/router/src/core/payments.rs [hunk 2]
```diff
@@ -1027,8 +1035,9 @@ impl<Ctx: PaymentMethodRetrieve> PaymentRedirectFlow<Ctx> for PaymentRedirectSyn
         ))
         .await
     }
-    fn generate_response(
+    async fn generate_response(
```
[the remark this answers is not in the document]
Comment (sai-harsha-vardhan) [reply crates/router/src/core/payments.rs:1038]: We need to fetch the poll configs for a connector, to send it in the response

[the remark this answers is not in the document]
Comment (hrithikesh026) [reply crates/router/src/core/payments.rs:1038]: I think you can accept a generic type in fn generate_response() and That generic type would be PollConfigs for PaymentAuthenticateCompleteAuthorize.

File: crates/router/src/core/payments.rs [hunk 3]
```diff
@@ -1093,51 +1130,69 @@ impl<Ctx: PaymentMethodRetrieve> PaymentRedirectFlow<Ctx> for PaymentAuthenticat
...
+            .store
+            .find_config_by_key_unwrap_or(
+                &format!("poll_config_external_three_ds_{connector}"),
+                Some(default_config_str),
+            )
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("The poll config was not found in the DB")?;
+        let poll_config =
+            serde_json::from_str::<Option<router_types::PollConfig>>(&poll_config.config)
```
Comment (Narayanbhat166) [inline crates/router/src/core/payments.rs:1155]: can you use parse struct here?

**Review by Narayanbhat166 — APPROVED** [review]

File: crates/router/src/core/payments.rs (hunk 3 above)
Comment (sai-harsha-vardhan) [reply crates/router/src/core/payments.rs:1155]: Will take that up in next PR

**Review by hrithikesh026 — APPROVED** [review]

**Pull request merged**